pkg/tuple/tuple.go

```diff
@@ -263,6 +271,11 @@ func Delete(tpl *core.RelationTuple) *core.RelationTupleUpdate {
 	}
 }
 
+func Equal(lhs, rhs *core.RelationTuple) bool {
+	// TODO(jschorr): Use a faster method then string comparison for caveats.
+	return OnrEqual(lhs.ResourceAndRelation, rhs.ResourceAndRelation) && OnrEqual(lhs.Subject, rhs.Subject) && MustStringCaveat(lhs.Caveat) == MustStringCaveat(rhs.Caveat)
```

Comment:
We still stringify the caveat if everything else matches. There might be some optimizations that could be made here. For example, we could count the keys before stringifying. I'm not sure if this is worth doing.

Reply:
There is a function in the codebase that computes a hash of the caveat context. I'm not sure that'd be faster than stringifying tho.